### PR TITLE
added example that uses system fields at ingestion

### PR DESCRIPTION
**Pull request by sergioferragut**
Added section on this topic to the batch ingestion notebook.

### CODE REVIEW

File: notebooks/02-ingestion/02-batch-ingestion.ipynb [hunk 1]
```diff
@@ -976,7 +1065,8 @@
     "druid.datasources.drop('kttm-nested-batch', True)\n",
     "druid.datasources.drop('kttm-enhanced-batch', True)\n",
     "druid.datasources.drop('wiki-users-batch', True)\n",
```
Comment (petermarshallio) [inline notebooks/02-ingestion/02-batch-ingestion.ipynb:1067]: These table names don't conform to the standard - e.g. should be example-wikipedia-{etc}

File: notebooks/02-ingestion/02-batch-ingestion.ipynb [hunk 2]
```diff
@@ -631,6 +631,95 @@
     "\"\"\")"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "3e330808-696b-4278-be8e-269924cc8089",
+   "metadata": {},
+   "source": [
+    "## NEW in Druid 29.0 - System Fields for Batch Ingestion\n",
```
Comment (petermarshallio) [inline notebooks/02-ingestion/02-batch-ingestion.ipynb:639]: As per elsewhere, avoid talking about specific versions in the body, except for the intro where the version requirement is given. Then the old notebook gets a version number suffix, as with lookups in 28,

File: notebooks/02-ingestion/02-batch-ingestion.ipynb [hunk 3]
```diff
@@ -631,6 +631,95 @@
     "\"\"\")"
    ]
   },
+  {
```
Comment (petermarshallio) [inline notebooks/02-ingestion/02-batch-ingestion.ipynb:634]: A note that this notebook has the SQL keywords in `CODE` though we've been asked to keep them capitalized but note in `pre` any more. Maybe a good time to update that if you have the time :D

File: notebooks/02-ingestion/02-batch-ingestion.ipynb [hunk 4]
```diff
@@ -631,6 +631,95 @@
...
+    "    )\n",
+    "```\n",
+    "\n",
+    "and in the EXTEND clause add the fields so they are accesible to SELECT :\n",
+    "```\n",
+    "EXTEND (\"__file_uri\" VARCHAR,\"__file_path\" VARCHAR, ...)\n",
+    "\n",
+    "```\n",
+    "\n",
+    "This example ingests three files (20 million rows each) into a roll-up table to demonstrate the use of system fields. It takes a while, so be patient:"
```
Comment (petermarshallio) [inline notebooks/02-ingestion/02-batch-ingestion.ipynb:664]: Unfortunately I couldn't complete the review here as I get:

```
[UnknownError](https://druid.apache.org/docs/latest/multi-stage-query/reference.html#error_UnknownError): java.lang.RuntimeException: java.lang.ClassCastException: class org.apache.druid.segment.transform.TransformedInputRow cannot be cast to class org.apache.druid.data.input.ListBasedInputRow (org.apache.druid.segment.transform.TransformedInputRow and org.apache.druid.data.input.ListBasedInputRow are in unnamed module of loader 'app') (Stack trace)

Failed task ID: query-457bc7e2-20fe-42c8-9c6e-39c87710ce47-worker0_0 (on host: 172.21.0.10:8101)
Debug: get query detail archive
```

**Review by petermarshallio — CHANGES_REQUESTED** [review]
Just (as with others) a note on the version header and line 44 again - but otherwise I wasn't able to complete the review as I got a Druid error.

File: notebooks/02-ingestion/02-batch-ingestion.ipynb (hunk 1 above)
Comment (sergioferragut) [reply notebooks/02-ingestion/02-batch-ingestion.ipynb:1067]: yeah, this was an earlier notebook. I'll change them all.

File: notebooks/02-ingestion/02-batch-ingestion.ipynb (hunk 3 above)
Comment (sergioferragut) [reply notebooks/02-ingestion/02-batch-ingestion.ipynb:634]: done

**Review by sergioferragut — COMMENTED** [review]
Let's talk about versioning, we don't seem to be on the same page.
I've done the rest of the suggestions.

File: notebooks/02-ingestion/02-batch-ingestion.ipynb (hunk 2 above)
Comment (sergioferragut) [reply notebooks/02-ingestion/02-batch-ingestion.ipynb]: Applied new versioning strategy in the latest commit.

**Pull request merged**